cli/defenseclaw/scanner/mcp.py

```
@@ -837,6 +841,26 @@ def _inspect_to_llm(il: InspectLLMConfig) -> LLMConfig:
     )
 
 
+def _scope_network_analyzer_dns(scanner: object) -> None:
+    """Keep analyzer traffic separate from the pinned MCP transport."""
+    for attr_name in ("_api_analyzer", "_llm_analyzer"):
+        analyzer = getattr(scanner, attr_name, None)
+        analyze = getattr(analyzer, "analyze", None)
+        if not callable(analyze):
+            continue
+
+        async def run_analyzer(*args, _analyze=analyze, **kwargs):
```

Ruff (0.16.0)

[warning] Missing return type annotation for private function `run_analyzer`

(ANN202)